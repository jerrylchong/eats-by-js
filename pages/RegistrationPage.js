import React, { useState } from 'react';
import {StatusBar, StyleSheet, TouchableOpacity, SafeAreaView, TextInput, KeyboardAvoidingView, Text, View, Platform} from "react-native";

function RegistrationPage({ navigation }) {
    const [username, setUsername] = useState('');
    const [password, setPassword] = useState('');
    const backHandler = () => {
        return (
            navigation.goBack()
        )
    }
    const submitHandler = () => {
        return (
            navigation.goBack()
        )
    }
    return (
        <SafeAreaView style = {styles.container}>
            <Text style = {styles.header}>Sign Up</Text>
            <KeyboardAvoidingView behavior={Platform.OS == "ios" ? "padding" : "height"} style = {styles.list}>
                <TextInput
                    style={styles.input}
                    placeholder="Name"
                    onChangeText={(text) => {setTitle(text)}}
                    value={title}/>
                <TextInput
                    style={styles.input}
                    placeholder="Description"
                    onChangeText={(text) => {setDesc(text)}}
                    value={desc}/>
            </KeyboardAvoidingView>
            <View style = {styles.buttonShadow}>
                <TouchableOpacity style = {styles.button} onPress = {submitHandler}>
                    <Text style = {styles.buttonText}>Submit</Text>
                </TouchableOpacity>
            </View>
            <View style = {styles.buttonShadow}>
                <TouchableOpacity style = {styles.button} onPress = {backHandler}>
                    <Text style = {styles.buttonText}>Back</Text>
                </TouchableOpacity>
            </View>
        </SafeAreaView>
    )
}

export default RegistrationPage;

const styles = StyleSheet.create({
    container: {
        flex: 1,
        flexDirection: 'column',
        backgroundColor: '#fff',
        alignItems: 'center',
        marginTop: StatusBar.currentHeight,
    },
    header: {
        fontSize: 20,
        marginTop: '10%'
    },
    list: {
        width: '80%',
        height: '80%',
        alignItems: 'center',
        justifyContent: 'space-evenly',
    },
    inputHeader: {
        flexDirection: 'row',
        alignSelf: 'flex-start'
    },
    input: {
        borderBottomWidth: 1,
        borderColor: 'black',
        fontSize: 12,
        width: '100%',
        height: '5%',
        paddingHorizontal: 5,
    },
    buttonShadow: {
        width: '25%',
        height: '5%',
        backgroundColor: 'white',
        borderRadius: 20,
        shadowColor: "#000",
        shadowOffset: {
            width: 0,
            height: 0,
        },
        shadowOpacity: 0.25,
        shadowRadius: 3.84,
        elevation: 2,
    },
    button: {
        height: '100%',
        width: '100%',
        alignItems: 'center',
        justifyContent: 'center',
        paddingHorizontal: 20,
        paddingVertical: 3,
    },
    buttonText: {
        fontSize: 12,
        color: 'black',
    }
})