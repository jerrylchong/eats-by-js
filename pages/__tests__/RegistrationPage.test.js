import React from 'react';
import renderer from 'react-test-renderer';
import Enzyme, {shallow} from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import RegistrationPage from "../RegistrationPage";
import {SafeAreaProvider} from "react-native-safe-area-context";

// snapshot test
test('renders correctly', () => {
    const tree = renderer.create(
        <SafeAreaProvider initialSafeAreaInsets={{ top: 0, left: 0, right: 0, bottom: 0 }}>
            <RegistrationPage
                navigation={{navigate: jest.fn()}}
            />
        </SafeAreaProvider>
    ).toJSON();
    expect(tree).toMatchSnapshot();
})

Enzyme.configure({ adapter: new Adapter() });