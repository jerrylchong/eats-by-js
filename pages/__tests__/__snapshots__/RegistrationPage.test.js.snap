// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<RNCSafeAreaView
  onInsetsChange={[Function]}
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Array [
        Object {
          "alignItems": "center",
          "backgroundColor": "#fff",
          "flex": 1,
          "flexDirection": "column",
        },
        Object {
          "paddingLeft": 0,
          "paddingRight": 0,
          "paddingTop": 0,
        },
      ]
    }
  >
    <View
      accessibilityIgnoresInvertColors={true}
      style={
        Object {
          "bottom": 0,
          "height": 327.33812949640287,
          "position": "absolute",
          "width": "100%",
        }
      }
    >
      <Image
        source={1}
        style={
          Array [
            Object {
              "bottom": 0,
              "left": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
            },
            Object {
              "height": 327.33812949640287,
              "width": "100%",
            },
            undefined,
          ]
        }
      />
    </View>
    <Text
      style={
        Object {
          "alignSelf": "flex-start",
          "color": "#404040",
          "fontFamily": "Ubuntu-Bold",
          "fontSize": 26,
          "marginLeft": 112.5,
          "position": "relative",
          "top": "20%",
        }
      }
    >
      Create an Account
    </Text>
    <View
      onLayout={[Function]}
      style={
        Array [
          Object {
            "alignItems": "center",
            "height": "15%",
            "position": "relative",
            "top": "30%",
            "width": "100%",
          },
          Object {
            "paddingBottom": 0,
          },
        ]
      }
    >
      <TextInput
        allowFontScaling={true}
        autoCapitalize="none"
        onChangeText={[Function]}
        placeholder="Username"
        placeholderTextColor="#404040"
        rejectResponderTermination={true}
        style={
          Object {
            "backgroundColor": "#d9d9d9",
            "borderRadius": 262.5,
            "color": "#404040",
            "fontFamily": "Ubuntu",
            "fontSize": 12,
            "fontWeight": "normal",
            "height": 82.5,
            "paddingHorizontal": "5%",
            "position": "relative",
            "width": 562.5,
          }
        }
        textContentType="username"
        underlineColorAndroid="transparent"
        value=""
      />
      <TextInput
        allowFontScaling={true}
        autoCapitalize="none"
        onChangeText={[Function]}
        placeholder="Password"
        placeholderTextColor="#404040"
        rejectResponderTermination={true}
        secureTextEntry={true}
        style={
          Object {
            "backgroundColor": "#d9d9d9",
            "borderRadius": 262.5,
            "color": "#404040",
            "fontFamily": "Ubuntu",
            "fontSize": 12,
            "fontWeight": "normal",
            "height": 82.5,
            "paddingHorizontal": "5%",
            "position": "relative",
            "top": "15%",
            "width": 562.5,
          }
        }
        textContentType="newPassword"
        underlineColorAndroid="transparent"
        value=""
      />
    </View>
    <View
      style={
        Object {
          "backgroundColor": "#ff6961",
          "borderRadius": 20,
          "elevation": 5,
          "height": 66.7,
          "margin": 10,
          "position": "relative",
          "shadowColor": "#000",
          "shadowOffset": Object {
            "height": 2,
            "width": 0,
          },
          "shadowOpacity": 0.25,
          "shadowRadius": 3.84,
          "top": "40%",
          "width": 333.5,
        }
      }
    >
      <View
        accessible={true}
        focusable={true}
        isTVSelectable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "height": "100%",
            "justifyContent": "center",
            "opacity": 1,
            "width": "100%",
          }
        }
      >
        <Text
          style={
            Object {
              "color": "white",
              "fontFamily": "Ubuntu",
            }
          }
        >
          Create Account
        </Text>
      </View>
    </View>
    <View
      style={
        Object {
          "alignItems": "center",
          "height": "20%",
          "position": "relative",
          "top": "50%",
          "width": "100%",
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#404040",
            "fontFamily": "Ubuntu",
          }
        }
      >
        Already have an account?
      </Text>
      <View
        accessible={true}
        focusable={true}
        isTVSelectable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
      >
        <Text
          style={
            Object {
              "color": "#ff8041",
              "fontFamily": "Ubuntu-Medium",
              "margin": "5%",
            }
          }
        >
          Sign In
        </Text>
      </View>
    </View>
  </View>
</RNCSafeAreaView>
`;
